In [1]:
import pandas as pd

In [2]:
sales = pd.read_csv("sales.csv")

In [7]:
%%sql
SELECT
    strftime('%Y-%m', CAST(date AS DATE)) AS month,
    SUM(total) AS total_sales
FROM
    sales
WHERE
    CAST(date AS DATE) BETWEEN '2021-06-01' AND '2021-08-31'
GROUP BY
    month
ORDER BY
    month;

month  total_sales
0  2021-06     95320.03
1  2021-07     93547.91
2  2021-08    100245.06

In [8]:
%%sql
SELECT
    strftime('%Y-%m', CAST(date AS DATE)) AS month,
    strftime('%W', CAST(date AS DATE)) AS week,
    SUM(total) AS total_sales
FROM
    sales
WHERE
    CAST(date AS DATE) BETWEEN '2021-06-01' AND '2021-08-31'
GROUP BY
    month, week
ORDER BY
    month, week;

month week  total_sales
0   2021-06   22     15827.97
1   2021-06   23     19744.15
2   2021-06   24     26907.63
3   2021-06   25     20542.50
4   2021-06   26     12297.78
5   2021-07   26     16199.46
6   2021-07   27     23315.98
7   2021-07   28     19205.91
8   2021-07   29     15958.17
9   2021-07   30     18868.39
10  2021-08   30      4599.58
11  2021-08   31     33777.36
12  2021-08   32     25284.92
13  2021-08   33     20202.94
14  2021-08   34     16380.26

# Sales Trends Analysis

This report provides an analysis of sales trends based on the provided dataset for the period from June to August 2021.

## Overview

The sales trends analysis aims to understand how sales performed over time during the specified period. By examining the total sales for each month, we can identify patterns, fluctuations, and potential factors influencing sales.

## Findings

### Overall Sales Trends

- Total sales in June 2021 amounted to $95,320.03.
- Total sales in July 2021 amounted to $93,547.91.
- Total sales in August 2021 amounted to $100,245.06.

### Observations

1. **June to July Comparison:**
   - Total sales slightly decreased from June to July, indicating a minor downturn in sales during this period.

2. **July to August Comparison:**
   - Total sales increased from July to August, showing a notable uptick in sales during this period.

3. **Overall Trend:**
   - Despite the decrease in July, the overall trend from June to August is upward, with an increase in total sales over the three-month period.

4. **Possible Factors:**
   - Seasonal influences, marketing campaigns, economic conditions, or changes in consumer behavior could have contributed to the observed fluctuations in sales.

## Conclusion

The analysis of sales trends from June to August 2021 provides valuable insights into the performance of sales over time. Understanding these trends can aid in strategic decision-making, such as inventory management, pricing strategies, and marketing efforts, to optimize sales performance and drive business growth.